src/main/java/com/example/petbuddybackend/service/rating/RatingService.java

```diff
@@ -16,86 +15,118 @@
+    private static final String USER_CANNOT_RATE_HIMSELF_MESSAGE = "User cannot rate himself";
+    private static final String NOT_CLIENT_MESSAGE = "User %s is not a client of care %d";
 
     private final RatingRepository ratingRepository;
     private final CareService careService;
     private final UserService userService;
     private final RatingMapper ratingMapper = RatingMapper.INSTANCE;
 
     public Page<RatingResponse> getRatings(Pageable pageable, String caretakerEmail) {
-        return ratingRepository.findAllByCaretakerEmail(caretakerEmail, pageable)
+        return ratingRepository.findAllByCare_Caretaker_Email(caretakerEmail, pageable)
+                .map(this::refreshRatingPhotos)
                 .map(ratingMapper::mapToRatingResponse);
     }
 
+    @Transactional
     public RatingResponse rateCaretaker(String clientEmail, Long careId, int rating,
                                         String comment) {
 
         Care care = careService.getCareById(careId);
         String caretakerEmail = care.getCaretaker().getEmail();
-        assertCareBetweenCaretakerAndClientIsValid(clientEmail, care);
+        assertNotRatingSelf(clientEmail, caretakerEmail);
+        assertCareOfClient(care, clientEmail);
+        assertPaidState(care);
 
-        if(caretakerEmail.equals(clientEmail)) {
-            throw new IllegalActionException("User cannot rate himself");
-        }
-
-        return ratingMapper.mapToRatingResponse(
-                createOrUpdateRating(caretakerEmail, clientEmail, careId, rating, comment)
-        );
+        Rating ratingEntity = createOrUpdateRating(caretakerEmail, clientEmail, careId, rating, comment);
+        refreshRatingPhotos(ratingEntity);
+        return ratingMapper.mapToRatingResponse(ratingEntity);
     }
 
     public RatingResponse deleteRating(String clientEmail, Long careId) {
-        Care care = careService.getCareById(careId);
-        String caretakerEmail = care.getCaretaker().getEmail();
-
-        RatingKey ratingKey = new RatingKey(caretakerEmail, clientEmail, careId);
-        Rating rating = getRating(ratingKey);
-        ratingRepository.deleteById(ratingKey);
-
+        Rating rating = getRatingOfClient(careId, clientEmail);
+        ratingRepository.delete(rating);
+        refreshRatingPhotos(rating);
         return ratingMapper.mapToRatingResponse(rating);
     }
 
+    private Rating refreshRatingPhotos(Rating rating) {
+        userService.renewProfilePicture(rating.getCare().getClient().getAccountData());
+        return rating;
+    }
+
     private Rating createOrUpdateRating(String caretakerEmail, String clientEmail, Long careId, int rating,
                                         String comment) {
-        Rating ratingEntity = getOrCreateRating(caretakerEmail, clientEmail, careId);
+
+        Care care = careService.getCareById(careId);
+        assertNotRatingSelf(clientEmail, caretakerEmail);
```

Comment:
sprawdzamy to juz w metodzie, ktora tą metode wywoluje, wiec nie sprawdzalby drugi raz

Reply:
Szczerze to może nawet nie trzeba tego sprawdzać, bo opieka ma `client_email <> caretaker_email` no i sprawdzam czy principal jest klientem w opiece